lib/mail/message.rb

```diff
@@ -2063,6 +2063,8 @@ def add_encoding_to_body
     def identify_and_set_transfer_encoding
         if body && body.multipart?
             self.content_transfer_encoding = @transport_encoding
+        elsif self.mime_type == 'message/rfc822'
+            self.content_transfer_encoding = Mail::Encodings::Binary
```

Comment:
Ideally we negotiate the best of 7bit, 8bit, binary. Can we push this decisionmaking down into the existing content transfer encoding negotiations, e.g. by passing an array of allowed encodings to `body.get_best_encoding`?